papi/handlers.go

```diff
@@ -463,6 +463,105 @@ func (h handler) GetMonthlyInfosByYear(c echo.Context) error {
 	return c.JSON(http.StatusOK, sumMI)
 }
 
+func (h handler) V2GetAggregateIndexes(c echo.Context) error {
+	jurisdicao := c.QueryParam("jurisdicao")
+	orgao := c.QueryParam("orgao")
+	agregado := c.QueryParam("agregado")
+
+	groupMap := map[string]string{
+		"justica-eleitoral":    "Eleitoral",
+		"ministerios-publicos": "Ministério",
+		"justica-estadual":     "Estadual",
+		"justica-do-trabalho":  "Trabalho",
+		"justica-federal":      "Federal",
+		"justica-militar":      "Militar",
+		"justica-superior":     "Superior",
+		"conselhos-de-justica": "Conselho",
+	}
+
+	var indexes map[string][]models.IndexInformation
+	var err error
+
+	// Verificamos se a jurisdição foi informada e se é válida
+	if groupName, ok := groupMap[jurisdicao]; ok {
+		indexes, err = h.client.Db.GetIndexInformation(groupName)
+		if err != nil {
+			return c.JSON(http.StatusBadRequest, fmt.Sprintf("Erro consultando os índices para o grupo %s.", groupName))
+		}
+		// Verificamos se o órgão foi informado
+	} else if orgao != "" {
+		indexes, err = h.client.Db.GetIndexInformation(orgao)
+		if err != nil {
+			return c.JSON(http.StatusBadRequest, fmt.Sprintf("Erro consultando os índices para o órgão %s.", orgao))
+		}
+	} else {
+		indexes, err = h.client.Db.GetAllIndexInformation()
+		if err != nil {
+			return c.JSON(http.StatusBadRequest, fmt.Sprintf("Erro consultando os índices de todos os órgãos."))
+		}
+	}
+	indexMap := make(map[string][]indexInformation)
+	aggregateScore := make(map[string]float64)
+	aggregateEasinessScore := make(map[string]float64)
+	aggregateCompletenessScore := make(map[string]float64)
+
+	for id, index := range indexes {
+		for _, a := range index {
+			indexMap[id] = append(indexMap[id], indexInformation{
+				Month: a.Month,
+				Year:  a.Year,
+				Score: &score{
+					Score:             a.Score.Score,
+					EasinessScore:     a.Score.EasinessScore,
+					CompletenessScore: a.Score.CompletenessScore,
+				},
+				Metadata: &metadata{
+					OpenFormat:       a.Meta.OpenFormat,
+					Access:           a.Meta.Access,
+					Extension:        a.Meta.Extension,
+					StrictlyTabular:  a.Meta.StrictlyTabular,
+					ConsistentFormat: a.Meta.ConsistentFormat,
+					HasEnrollment:    a.Meta.HaveEnrollment,
+					HasCapacity:      a.Meta.ThereIsACapacity,
+					HasPosition:      a.Meta.HasPosition,
+					BaseRevenue:      a.Meta.BaseRevenue,
+					OtherRecipes:     a.Meta.OtherRecipes,
+					Expenditure:      a.Meta.Expenditure,
+				},
+			})
+			aggregateScore[id] += a.Score.Score
+			aggregateCompletenessScore[id] += a.Score.CompletenessScore
+			aggregateEasinessScore[id] += a.Score.EasinessScore
+		}
+	}
+	var aggregate []aggregateIndexes
+	for id, index := range indexMap {
+		aggregateScore[id] = aggregateScore[id] / float64(len(index))
```

Comment:
Note que você só precisa do `len(index)` para calcular os agregados e os somatórios para calcular os agregados. 

Note também que essa vai ser a chamada realizada para renderizar a página inicial do projeto. Ou seja, ela vai ser chamada diversas vezes.

Dito isso, minha sugestão é você apenas calcular o agregado quando o parâmetro for verdadeiro. Isso vai evitar que  a memória necessária para armazenar o detalhamento seja solicitada, usada e depois jogada fora. Isso impacta muito a forma que o coletor de lixo opera.